corpdir-srv-sync/src/main/java/com/github/bordertech/corpdir/sync/service/ImportServiceAction.java

```diff
@@ -0,0 +1,133 @@
+package com.github.bordertech.corpdir.sync.service;
+
+import com.github.bordertech.corpdir.api.common.ApiIdObject;
+import com.github.bordertech.corpdir.api.response.DataResponse;
+import com.github.bordertech.corpdir.api.v1.VersionCtrlService;
+import com.github.bordertech.corpdir.api.v1.model.VersionCtrl;
+import com.github.bordertech.corpdir.sync.common.DataSynchronisation;
+import com.github.bordertech.corpdir.sync.common.DataVersionSynchronisation;
+import com.github.bordertech.corpdir.sync.impl.ContactSynchronisation;
+import com.github.bordertech.corpdir.sync.impl.LocationSynchronisation;
+import com.github.bordertech.corpdir.sync.impl.OrgUnitSynchronisation;
+import com.github.bordertech.corpdir.sync.impl.PositionSynchronisation;
+import com.github.bordertech.corpdir.sync.impl.PositionTypeSynchronisation;
+import com.github.bordertech.corpdir.sync.impl.UnitLevelSynchronisation;
+import com.github.bordertech.didums.Didums;
+import com.github.bordertech.taskmaster.service.ServiceAction;
+import java.util.Date;
+import java.util.List;
+import org.apache.commons.lang.time.DateFormatUtils;
+
+/**
+ *
+ * @author exiqaj
+ */
+public class ImportServiceAction implements ServiceAction<String, ApiIdObject> {

```

Comment:
What if the `sync` logic has its own `interface` which the `ServiceAction` could just call an implementation. This does not tie the `sync` logic to `TaskMaster` as the `TaskMaster` is just a means of invoking the `sync`

The `ServiceAction` could just be in the `UI` module.

Reply:
Ok, moved `sync` logic into own `interface`, and moved `ServiceAction` into `View` tier